Pip's name key can now accept remote package address.

yeukhon opened the pull request:
**Summary:**

Pip module would abort when name is a remote package address because
the module was expecting a version if `=` is part of the name value.

Furthermore, the pip module would require either name or requirement to
be a key, although the documentation table said neither was required.
The fact that one of them must be present is not documented in the
documentation leads to confusion. This commit added this fact as part
of description.

In this commit, we resolve the confusion by stating either `name`
or `requirement` is needed. Next, if the user puts remote address
as the value of the `name` key, we will not use mirror. Lastly,
if the user uses the remote serivice address as the name of the
package and the user does not supply -e option in extra_vars
(which is the whole point of this commit), we will add -e to
 extra_vars so pip command can run with -e option.

yeukhon committed:
Pip's name key can now accept remote package address.

Summary:

Pip module would abort when name is a remote package address because
the module was expecting a version if ``=`` is part of the name value.

Furthermore, the pip module would require either name or requirement to
be a key, although the documentation table said neither was required.
The fact that one of them must be present is not documented in the
documentation leads to confusion. This commit added this fact as part
of description.

In this commit, we resolve the confusion by stating either ``name``
or ``requirement`` is needed. Next, if the user puts remote address
as the value of the ``name`` key, we will not use mirror. Lastly,
if the user uses the remote serivice address as the name of the
package and the user does not supply -e option in extra_vars
(which is the whole point of this commit), we will add -e to
 extra_vars so pip command can run with -e option.

mpdehaan commented:
Does it make sense to also add code to support tarball installation from pip at this time?

yeukhon commented:
@mpdehaan  I haven't tried with tarball. I can give it a try though. I wasn't aware of tarball.

yeukhon committed:
Pip can now accept vcs or tarball as name. Prevented adding mirrors when name is vcs.

**Summary**:

There was a bug in the previous commit; pip module would add --use-mirrors options to a source package when state is absent. The bug is resolved in this commit by checking ``not is_package`` in the if branch.

Furthermore, in order to support non-vcs source name like tarballs, we must not add -e option to the arg list. Given this circumstance, this commit have is_tar and is_vcs and the latter is checked to add -e option. Since mirrors do not make sense with vcs or tarball source, this commit will not add --use-mirrors (default to True) as always.

yeukhon commented:
Okay. I have just tested this commit should work. Now pip can accept vcs, tarball or regaulr pkg name.

mpdehaan commented:
merged, thanks!
